.github/workflows/ci-complete.yml

```diff
@@ -44,11 +60,48 @@ jobs:
         with:
           java-version: ${{ matrix.java }}
           develocity-access-key: ${{ secrets.GE_ACCESS_TOKEN }}
-      - uses: actions/download-artifact@v4
+      - name: Download build scan archive
+        id: download-build-scan
+        uses: actions/download-artifact@v4
+        continue-on-error: true
         with:
           github-token: ${{ github.token }}
           run-id: ${{ github.event.workflow_run.id }}
           name: build-scan-test-${{ matrix.java }}
-          path: ~/.gradle/build-scan-data
-      - name: Publish scan
-        run: ./gradlew --info buildScanPublishPrevious
+          path: ~/.gradle/build-scan-data  # This is where Gradle buffers unpublished build scan data when --no-scan is given
+      - name: Handle missing scan
+        if: ${{ steps.download-build-scan.outcome == 'failure' }}
+        run: |
+          echo "Could not download build scans from ${{ github.event.workflow_run.html_url }} " >> $GITHUB_STEP_SUMMARY
+      - name: Publish Scan
+        id: publish-build-scan
+        continue-on-error: true
+        if: ${{ steps.download-build-scan.outcome == 'success' }}
+        run: |
+          ./gradlew --info buildScanPublishPrevious > gradle.out
```

Comment:
Do we still need to publish build scan in the `Compile and validate` phase? It can't work with public fork and we already collect all PRs' scan, so maybe we can remove the `--scan` from `Compile and validate` phase

https://github.com/apache/kafka/blob/trunk/.github/workflows/build.yml#L66

Reply:
We can keep the `--scan` for trunk builds and use `--no-scan` for PRs. I'll update that.

Reply:
You can see the `--no-scan` working here https://github.com/apache/kafka/actions/runs/10779482765/job/29893102166?pr=17140#step:6:3